concrete-core/src/backends/core/private/crypto/bootstrap/mod.rs

```diff
@@ -3,116 +3,70 @@
-    /// // accumulator is a trivial encryption of [0, 1/2N, 2/2N, ...]
-    /// let mut accumulator =
-    ///     GlweCiphertext::allocate(0u32, polynomial_size, rlwe_dimension.to_glwe_size());
-    /// accumulator
-    ///     .get_mut_body()
-    ///     .as_mut_tensor()
-    ///     .iter_mut()
-    ///     .enumerate()
-    ///     .for_each(|(i, a)| {
-    ///         *a = (i as f64 * 2_f64.powi(32_i32 - 10 - 1)).cast_into();
-    ///     });
-    ///
-    /// // bootstrap
-    /// fourier_bsk.bootstrap(&mut lwe_out, &lwe_in, &accumulator);
-    /// ```
-    fn bootstrap<C1, C2, C3>(
-        &self,
-        lwe_out: &mut LweCiphertext<C1>,
-        lwe_in: &LweCiphertext<C2>,
-        accumulator: &GlweCiphertext<C3>,
-    ) where
-        LweCiphertext<C1>: AsMutTensor<Element = Self::CiphertextScalar>,
-        LweCiphertext<C2>: AsRefTensor<Element = Self::CiphertextScalar>,
-        GlweCiphertext<C3>: AsRefTensor<Element = Self::CiphertextScalar>;
+// mod surrogate;
+
+#[derive(Debug, Clone)]
+pub struct FftBuffers {
+    // The fft plan is stored here. This way, we don't pay the price of allocating it every
+    // time we need to bootstrap with the same key.
+    fft: Fft,
+    // The buffers used to perform the fft are also stored in the bootstrap key. Again, the same
+    // logic apply, and we don't have to allocate them multiple times.
+    first_buffer: FourierPolynomial<AlignedVec<Complex64>>,
+    second_buffer: FourierPolynomial<AlignedVec<Complex64>>,
+    output_buffer: Tensor<AlignedVec<Complex64>>,
+}
+
+#[derive(Debug, Clone)]
+pub struct FourierBskBuffers<Scalar> {
+    // Those buffers are also used to store the lut and the rounded input during the bootstrap.
+    lut_buffer: GlweCiphertext<Vec<Scalar>>,
+    rounded_buffer: GlweCiphertext<Vec<Scalar>>,
+    fft_buffers: FftBuffers,
+}
+
+impl<Scalar> FourierBskBuffers<Scalar>
+where
+    Scalar: UnsignedTorus,
+{
+    pub fn for_key<Key: LweBootstrapKeyEntity>(key: &Key) -> Self {
+        let poly_size = key.polynomial_size();
+        let glwe_size = key.glwe_dimension().to_glwe_size();
+        Self::new(poly_size, glwe_size)
+    }
+
+    pub fn new(poly_size: PolynomialSize, glwe_size: GlweSize) -> Self {
+        let fft = Fft::new(poly_size);
+        let first_buffer = FourierPolynomial::allocate(Complex64::new(0., 0.), poly_size);
+        let second_buffer = FourierPolynomial::allocate(Complex64::new(0., 0.), poly_size);
+        let output_buffer = Tensor::from_container(AlignedVec::new(poly_size.0 * glwe_size.0));
+        let lut_buffer = GlweCiphertext::allocate(Scalar::ZERO, poly_size, glwe_size);
+        let rounded_buffer = GlweCiphertext::allocate(Scalar::ZERO, poly_size, glwe_size);
+
+        Self {
+            lut_buffer,
+            rounded_buffer,
+            fft_buffers: FftBuffers {
+                fft,
+                first_buffer,
+                second_buffer,
+                output_buffer,
+            },
+        }
+    }
```

Comment:
Wouldn't it be better to put that in a `fourier/buffers.rs` file ? This is specific to the `FourierBootstrapKey` so at least it looks like it belongs to the `fourier` module to me. Maybe you had something different in mind ?

Reply:
At first I put it there because it was needed in the `Bootstrap` trait definition, but since that trait was removed it makes sense to move this to a better place